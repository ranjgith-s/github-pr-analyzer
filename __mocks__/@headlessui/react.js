// __mocks__/@headlessui/react.js
const React = require('react');

function passthrough({ children, ...props }) {
  return React.createElement('div', props, children);
}

module.exports = {
  __esModule: true,
  Card: passthrough,
  CardHeader: passthrough,
  CardBody: passthrough,
  CardFooter: passthrough,
  HeroUIProvider: passthrough,
};
