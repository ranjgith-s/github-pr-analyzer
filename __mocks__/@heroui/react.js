const React = require('react');

// List of custom props to filter out for DOM elements
const CUSTOM_PROPS = [
  'isCurrent',
  'endContent',
  'onPress',
  'variant',
  'color',
  'size',
  'underline',
  'src',
  'alt',
  'className',
  'children',
];

function filterProps(props) {
  const domProps = { ...props };
  CUSTOM_PROPS.forEach((key) => {
    if (key in domProps && key !== 'className' && key !== 'children') {
      delete domProps[key];
    }
  });
  return domProps;
}

function passthrough({ children, ...props }) {
  return React.createElement('div', { ...filterProps(props), className: props.className }, children);
}

function InputMock(props) {
  return React.createElement('input', filterProps(props));
}

function ButtonMock({ children, onPress, onClick, ...props }) {
  // Map onPress to onClick for DOM, do not pass onPress to DOM
  const handler = onPress || onClick;
  const domProps = { ...filterProps(props), onClick: handler, className: props.className };
  return React.createElement('button', domProps, children);
}

function Spinner() {
  // Only return a simple span for test, not a div with data-testid
  return React.createElement('span', null, 'Loading...');
}

function CardMock({ children, ...props }) {
  return React.createElement('div', { ...filterProps(props), className: (props.className || '') + ' mock-card' }, children);
}

module.exports = {
  __esModule: true,
  Card: CardMock,
  CardHeader: passthrough,
  CardBody: passthrough,
  CardFooter: passthrough,
  Button: ButtonMock,
  Input: InputMock,
  Avatar: passthrough,
  Breadcrumbs: passthrough,
  BreadcrumbItem: passthrough,
  Spinner,
};
