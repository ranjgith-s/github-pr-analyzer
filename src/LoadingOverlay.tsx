import React, { useEffect, useState } from 'react';

interface LoadingOverlayProps {
  show: boolean;
  messages: string[];
}

export default function LoadingOverlay({
  show,
  messages,
}: LoadingOverlayProps) {
  const [index, setIndex] = useState(0);

  useEffect(() => {
    if (!show) return;
    const id = setInterval(() => {
      setIndex((i) => (i + 1) % messages.length);
    }, 2000);
    return () => clearInterval(id);
  }, [show, messages.length]);

  if (!show) return null;

  return (
    <div
      data-testid="spinner"
      style={{
        position: 'fixed',
        top: 0,
        left: 0,
        width: '100%',
        height: '100%',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        flexDirection: 'column',
        background: 'rgba(248, 250, 252, 0.9)',
        opacity: 0.9,
        zIndex: 10,
      }}
    >
      <span style={{ fontSize: 32, marginBottom: 16 }}>‚è≥</span>
      <p
        style={{
          fontFamily: 'monospace',
          animation: 'pulse 1.5s ease-in-out infinite',
        }}
      >
        {messages[index]}
      </p>
    </div>
  );
}
