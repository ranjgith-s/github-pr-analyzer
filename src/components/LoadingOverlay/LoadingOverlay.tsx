import React, { useEffect, useState } from 'react';
import { Spinner, Card } from '@heroui/react';

interface LoadingOverlayProps {
  show: boolean;
  messages: string[];
}

export default function LoadingOverlay({
  show,
  messages,
}: LoadingOverlayProps) {
  const [index, setIndex] = useState(0);

  useEffect(() => {
    if (!show) return;
    const id = setInterval(() => {
      setIndex((i) => (i + 1) % messages.length);
    }, 2000);
    return () => clearInterval(id);
  }, [show, messages.length]);

  if (!show) return null;

  return (
    <div
      data-testid="spinner"
      className="fixed inset-0 z-50 flex flex-col items-center justify-center bg-background/80 backdrop-blur-sm"
      style={{ minHeight: '100vh' }}
    >
      <Card className="flex flex-col items-center gap-4 p-8 shadow-xl border border-divider bg-content1/90">
        <Spinner size="lg" color="primary" />
        <p className="text-base font-mono animate-pulse text-foreground/80 text-center min-w-[200px] min-h-[24px]">
          {messages[index]}
        </p>
      </Card>
    </div>
  );
}
