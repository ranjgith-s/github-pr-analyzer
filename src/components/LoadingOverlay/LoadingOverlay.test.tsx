import React from 'react';
import { render, screen, act } from '@testing-library/react';
import LoadingOverlay from './LoadingOverlay';

jest.useFakeTimers();

describe('LoadingOverlay', () => {
  const messages = ['Loading...', 'Almost there!', 'Just a moment'];

  it('renders spinner and first message when show is true', () => {
    render(<LoadingOverlay show={true} messages={messages} />);
    expect(screen.getByTestId('spinner')).toBeInTheDocument();
    const messageEls = screen.getAllByText(messages[0]);
    expect(messageEls.some((el) => el.tagName === 'P')).toBe(true);
  });

  it('renders nothing when show is false', () => {
    const { container } = render(
      <LoadingOverlay show={false} messages={messages} />
    );
    expect(container.firstChild).toBeNull();
  });

  it('cycles through messages', async () => {
    render(<LoadingOverlay show={true} messages={messages} />);
    let messageEls = screen.getAllByText(messages[0]);
    expect(messageEls.some((el) => el.tagName === 'P')).toBe(true);
    await act(async () => {
      jest.advanceTimersByTime(2000);
    });
    messageEls = await screen.findAllByText(messages[1]);
    expect(messageEls.some((el) => el.tagName === 'P')).toBe(true);
    await act(async () => {
      jest.advanceTimersByTime(2000);
    });
    messageEls = await screen.findAllByText(messages[2]);
    expect(messageEls.some((el) => el.tagName === 'P')).toBe(true);
    await act(async () => {
      jest.advanceTimersByTime(2000);
    });
    messageEls = await screen.findAllByText(messages[0]);
    expect(messageEls.some((el) => el.tagName === 'P')).toBe(true);
  });

  it('matches snapshot', () => {
    const { container } = render(
      <LoadingOverlay show={true} messages={messages} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
