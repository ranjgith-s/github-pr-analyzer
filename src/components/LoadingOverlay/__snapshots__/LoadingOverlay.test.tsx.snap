// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingOverlay matches snapshot 1`] = `
<div
  class="fixed inset-0 z-50 flex flex-col items-center justify-center bg-background/80 backdrop-blur-sm"
  data-testid="spinner"
  style="min-height: 100vh;"
>
  <div
    class="rounded-xl text-card-foreground flex flex-col items-center gap-4 p-8 shadow-xl border bg-card/90"
  >
    <span
      aria-label="Loading"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuetext="Loading"
      class="inline-flex items-center justify-center"
      role="progressbar"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-loader-circle animate-spin h-6 w-6 text-primary"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 12a9 9 0 1 1-6.219-8.56"
        />
      </svg>
    </span>
    <p
      class="text-base font-mono animate-pulse text-foreground/80 text-center min-w-[200px] min-h-[24px]"
    >
      Loading...
    </p>
  </div>
</div>
`;
